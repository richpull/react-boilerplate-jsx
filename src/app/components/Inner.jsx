import React, { Component } from 'react';
import Children from './Children';
import styled from 'styled-components';

const InnerContainer = styled.section`
  border: 1px solid #ccc;
  margin: 50px;
  padding: 50px;
`;

class Inner extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <InnerContainer>
        <h1>{this.props.title}</h1>
        <div>My css-{'>'} children.css</div>
        <Children />
      </InnerContainer>
    );
  }
}
export default Inner;
