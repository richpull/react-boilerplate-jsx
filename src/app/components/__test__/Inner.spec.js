import React from 'react';
import { shallow } from 'enzyme';
import Inner from '../Inner';
import renderer from 'react-test-renderer';

describe('>>>I N N E R --- Snapshot', () => {
  it('+++capturing Snapshot of Inner', () => {
    const renderedValue = renderer.create(<Inner title="Test title" />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
