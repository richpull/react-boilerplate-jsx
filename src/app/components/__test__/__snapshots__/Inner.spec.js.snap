// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>I N N E R --- Snapshot +++capturing Snapshot of Inner 1`] = `
<section
  className="sc-bdVaJa hmHvdz"
>
  <h1>
    Test title
  </h1>
  <div>
    My css-
    &gt;
     children.css
  </div>
  <div
    className="children"
  >
    <h1>
      I am Children Component
    </h1>
    <p>
      My css-
      &gt;
       styled component
    </p>
    <div
      className="test"
    >
      test.css
    </div>
  </div>
</section>
`;
